jsapp/js/components/apiTokenDisplay.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import bem from 'js/bem';
+import {dataInterface} from 'js/dataInterface';
+import TextBox from 'js/components/common/textBox';
+import Button from 'js/components/common/button';
+
+const HIDDEN_TOKEN_VALUE = '*'.repeat(40);
+
+/**
+ * Displays secret API token of a logged in user.
+ * The token is obfuscated until a "show me" button is clicked.
+ */
+export default function ApiTokenDisplay() {
+  const [token, setToken] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleTokenVisibility = () => {
+    setIsVisible(!isVisible);
+  };
+
+  useEffect(() => {
+    if (isVisible && token === null) {
+      const fetchToken = async () => {
+        setIsFetching(true);
+        try {
+          const result = await dataInterface.apiToken();
+          setToken(result.token);
+        } finally {
```

Comment:
I don't think this works, you need a catch even if the catch doesn't do anything.

Reply:
It does work, try in console: `try {throw 'foo'} finally {console.log('bar')} `

Reply:
It's valid syntax, but it's not ignoring the error. See how the uncaught error is present with finally but not with catch.

![image](https://user-images.githubusercontent.com/739307/195614000-877c276e-74a7-4f6d-bb70-3a1893a4a547.png)

[Further reading](https://stackoverflow.com/questions/5764107/is-try-without-catch-possible-in-javascript)